trulens_eval/trulens_eval/guardrails/base.py

```diff
@@ -34,10 +34,17 @@ def __call__(self, func):
 
         def wrapper(*args, **kwargs):
             contexts = func(*args, **kwargs)
+            
+            # TODO: this is a hack
+            main_input = None
+            for arg in args + tuple(kwargs.values()):
+                if isinstance(arg, str):
+                    main_input = arg
+                    break
             with ThreadPoolExecutor(max_workers=max(1, len(contexts))) as ex:
                 future_to_context = {
                     ex.submit(
-                        lambda context=context: self.feedback(args[1], context)
```

Comment:
@sfc-gh-jreini `args[1]` may not exist when only kwargs are given. also we may want to think of a better way of selecting the prompt string